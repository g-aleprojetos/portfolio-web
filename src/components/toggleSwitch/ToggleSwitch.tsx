import React from 'react';
import * as S from './ToggleSwitch.styles';

type Props = {
  isOn: boolean;
  handleOnclick: () => void;
};

export const ToggleSwitch = (props: Props) => {
  const {isOn, handleOnclick} = props;

  return (
    <S.Switch data-testid="toggleSwitch">
      <S.Input checked={isOn} onChange={handleOnclick} />
      <S.Span className={`slider round ${isOn ? 'checked' : ''}`}></S.Span>
    </S.Switch>
  );
};
