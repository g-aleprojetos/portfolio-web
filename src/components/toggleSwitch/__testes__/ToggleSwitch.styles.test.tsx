import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../ToggleSwitch.styles';

describe('ToggleSwitch.styles', () => {
  test('Input DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Input />);
    expect(container).toMatchSnapshot();
  });

  test('Span DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Span />);
    expect(container).toMatchSnapshot();
  });

  test('Switch DEVE ser igual ao snapshot ', () => {
    const {container} = render(<S.Switch />);
    expect(container).toMatchSnapshot();
  });
});
