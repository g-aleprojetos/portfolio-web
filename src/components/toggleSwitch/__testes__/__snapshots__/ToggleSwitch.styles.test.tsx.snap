// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch.styles Input DEVE ser igual ao snapshot 1`] = `
.c0 {
  opacity: 0;
  width: 0;
  height: 0;
}

.c0:checked+span {
  background-color: #15154E;
}

.c0:checked+span:before {
  transform: translateX(23.4px);
  background-color: #ffffff;
}

.c0:focus+span {
  box-shadow: 0 0 1px #15154E;
}

<div>
  <input
    class="c0"
    type="checkbox"
  />
</div>
`;

exports[`ToggleSwitch.styles Span DEVE ser igual ao snapshot 1`] = `
.c0 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #2196F3;
  transition: 0.4s;
  border-radius: 30.6px;
}

.c0:before {
  position: absolute;
  content: '';
  height: 23.4px;
  width: 23.4px;
  left: 3.6px;
  bottom: 3.6px;
  background-color: #FFE879;
  transition: 0.4s;
  border-radius: 50%;
}

<div>
  <span
    class="c0"
  />
</div>
`;

exports[`ToggleSwitch.styles Switch DEVE ser igual ao snapshot  1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 54px;
  height: 30.6px;
}

<div>
  <label
    class="c0"
  />
</div>
`;
